testcases/kernel/security/stack_clash/stack_clash_child.c

```diff
@@ -0,0 +1,93 @@
+/* File from https://bugzilla.novell.com/show_bug.cgi?id=CVE-2017-1000364 */
+
+#include <sys/mman.h>
+#include <sys/resource.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <alloca.h>
+#include <signal.h>
+#include <stdlib.h>
+
+#define TST_NO_DEFAULT_MAIN
+#include "tst_test.h"
+
+#define PAGE_SHIFT              12
+#define PAGE_SIZE               (1UL << PAGE_SHIFT)
+#define PAGE_MASK		(~(PAGE_SIZE-1))
+#define SIGNAL_STACK_SIZE	(1UL<<20)
+#define MEGABYTE		(1<<20)			
+#define THRESHOLD		(250 * PAGE_SIZE)
+
+void recurse(void)
+{
+	volatile char * ptr = alloca(1024);
+	*ptr = '\0';
+	recurse();
+}
+
+#define MAPPED_LEN PAGE_SIZE
+static unsigned long mapped_addr;
+void segv_handler(int sig, siginfo_t *info, void *data)
+{
+	unsigned long addr = (unsigned long)info->si_addr;
+	unsigned long mmap_end = mapped_addr + MAPPED_LEN;
+	unsigned long diff;
+
+	if (sig != SIGSEGV)
+		return;
+
+	(void) data;
+
+#ifndef CONFIG_STACK_GROWSUP
+	diff = addr - mmap_end;
+#else
+	diff = mmap_addr - addr;
+#endif
+
+	printf("mmap_end = %lx, addr = %lx\n", mmap_end, addr);
+	printf("%lx %lx\n", diff, THRESHOLD);
+	if (diff < THRESHOLD)
```

Comment:
My bad, I was thinking of more generic test where stack grows into existing mapping.